### DISCUSSION TITLE
Iterators and generators?

### DISCUSSION THREAD

**Discussion by jkleiser** [vlang/v#8034] [General]
Is there an easy way to do iterators and generators in V?
See JavaScript description here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators

**Comment by Delta456** [comment]
You can look at [this](https://github.com/vlang/v/blob/master/vlib/v/tests/for-in-iterator_test.v).

**Reply by jkleiser** [reply]
Will **coroutines** (when available) make it easier?

**Reply by Delta456** [reply]
I do not know.

**Reply by JalonSolov** [reply]
No, coroutines won't necessarily have anything to do with iterators and/or generators, since those don't have to be done in parallel.

**Reply by larpon** [reply]
> Will **coroutines** (when available) make it easier?

@jkleiser it is pretty easy already.
if you have a method on your struct named `next` that returns an optional value you have an iterator you can use in a for loop.

I don't think there's any plans for asynchronous iteration currently. If you have examples, suggestions or good usecases you could consider joining the discussions here:
https://github.com/vlang/v/discussions/7610
